src/Service.Tests/CosmosTests/QueryFilterTests.cs

```diff
@@ -642,6 +647,135 @@ public async Task TestFilterOnInnerNestedFields()
             await ExecuteAndValidateResult(_graphQLQueryName, gqlQuery, dbQuery);
         }
 
+        #region Field Level Auth
+        /// <summary>
+        /// Tests that the field level query filter succeeds requests when filter fileds are authorized
+        /// </summary>
+        [TestMethod]
+        public async Task TestQueryFilterFieldAuth_AuthorizedField()
+        {
+            string gqlQuery = @"{
+                earths(first: 1, " + QueryBuilder.FILTER_FIELD_NAME + @" : {id : {eq : """ + _idList[0] + @"""}})
```

Comment:
What if the filter field is authorized, but the query field is unauthorized e.g. here the return contains name too?

items {
id
name
}

Reply:
The query field auth is being indicated using ```@authorize``` on that field, and the schema field "read" is using for the Filter Parser. I don't see a way for the user to mark the query field "unauthorized" while having the schema lists that field as included? Let me know if I miss something?

Reply:
This is a great point! I verified as below. 

1. Limit the field auth to authenticated. 
![image](https://user-images.githubusercontent.com/40870773/236576207-26a3a70c-c201-4c99-b925-c70186589339.png)
2. Grand the field auth to anonymous. 
![image](https://user-images.githubusercontent.com/40870773/236576273-03b1333d-983c-422f-b258-85ffb45da890.png)
3. Query field is correctly guarded by hot chocolate.
![image](https://user-images.githubusercontent.com/40870773/236575991-beec5c12-f44a-4b98-aa9a-0358b798b6ce.png)
It looks like what happened is, is that the filter Parser gets called and then the hot chocolate auth field is guarded after.

Reply:
Thanks for the verification! Could you please add an automated test for this as well? Essentially, add the @authorize directive in the graphql schema used in the test and try to query for the field for the authorized role. 

I notice we have the @authorize directive added at the entity level i.e. `Moon`. Now that we are adding support for field level auth, I think its important to add this test.

Reply:
Added integration test for this scenario. :)

Reply:
I noted in another comment, you'll need to swap out the usage of `policy` and replace with `roles` in the `@authorize` directive. e.g. `@authorize(roles: "authenticated")`

Reply:
You are right, thank you for pointing this out! Updated with roles.